apps/www/_blog/2024-08-13-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,208 @@
+author: filipe
+image: 2024-08-13-supabase-realtime-broadcast-and-presence-authorization/OG.png
+thumb: 2024-08-13-supabase-realtime-broadcast-and-presence-authorization/thumb.png
+categories:
+tags:
+  - realtime
+  - engineering
+date: '2024-08-13'
+toc_depth: 2
+---
+
+Today we're releasing Realtime Authorization for Broadcast and Presence in Public Beta.
+
+For context, Supabase includes three useful extensions for building real-time applications.
+
+1. [Broadcast](https://supabase.com/docs/guides/realtime/broadcast): Send ephemeral, low-latency messages between users.
+2. [Presence](https://supabase.com/docs/guides/realtime/presence): Show when users are online and share state between users.
+3. [Postgres Changes](https://supabase.com/docs/guides/realtime/postgres-changes): Listen to Postgres database changes.
+
+This release introduces authorization for Broadcast and Presence using Row Level Security policies:
+
+```sql
+create policy "authenticated user can listen to all messages"
+on "realtime"."messages"
+as permissive for select
+to authenticated
+using ( true );
+```
+
+To facilitate authorization in Realtime, a new table `realtime.messages` is created automatically by Supabase Realtime when it connects to your Database. This table represents your [Channels](/docs/guides/realtime/concepts#channels) as objects against which you can write RLS policies.
+
+<Img
+  src={{
+    dark: '/images/blog/2024-08-13-supabase-realtime-broadcast-and-presence-authorization/schema--dark.png',
+    light:
+      '/images/blog/2024-08-13-supabase-realtime-broadcast-and-presence-authorization/schema--light.png',
+  }}
+  alt="The table that is used to test the policies."
+/>
+
+You can write Postgres RLS policies for this table and Realtime will use it them control the access to
+Broadcast and Presence for your channels based on the operations set in your policies:
+
+- `SELECT` policies - Able to receive messages from the channel
+- `INSERT` policies - Able to send messages to the channel
+
+## How it works without Authorization
+
+When you want to connect to a Realtime Channel, you do something close to the following code:
+
+```tsx
+// With authenticated user token
+client.realtime.setAuth(token)
+
+client.channel('channel').subscribe((status: string, err: any) => {
+  if (status === 'SUBSCRIBED') {
+    console.log('Connected')
+  }
+})
+```
+
+This will enable you to send messages but they will all be public. Any user would be subscribe to this Channel and read these messages if they had your anon token.
+
+## Adding Authorization to Realtime Channels
+
+To convert this into a Realtime Channel that checks RLS policies requires only two steps:
+
+- [Create RLS Policies](#create-rls-policies)
+- [Enabling Authorization on Channel subscription](#enabling-authorization-on-channel-subscription)
+
+### Create RLS policies
+
+Any RLS policy will work. Reference our [Github Discussion](github.com/orgs/supabase/discussions/22484) for more complex use cases.
```

Comment:
```suggestion
Reference our [Github Discussion](github.com/orgs/supabase/discussions/22484) for more complex use cases.
```

Reply:
Can move this to the bottom of `### Create RLS policies` section for better flow.